core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender43.java

```diff
@@ -71,15 +84,29 @@ public void dispose(HttpResponse response) {
     @Override
     public void close() {
         try {
-            httpClient.close();
+            ((CloseableHttpClient)getHttpClient()).close();
         } catch (IOException e) {
             InternalLogger.INSTANCE.error("Failed to close http client, exception: %s", e.toString());
         }
     }
 
     @Override
     public HttpClient getHttpClient() {
-        return httpClient;
+        if (!isClientInitialized) {
```

Comment:
First call will always find this condition true (isClientInitialized==false) and then it will "block" until the initializer thread has finished.

Reply:
This makes sense, just would like to know the scenario where this could happen. I was under impression that the channel first calls the constructor or initializer method to instantiate the instance of sender even before calling send().